src/cryptsetup/cryptsetup-generator.c

```diff
@@ -391,36 +396,47 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
                         return log_oom();
 
         } else if (streq(key, "luks.key")) {
+                size_t n;
+                _cleanup_free_ char *keyfile = NULL, *keydev = NULL;
+                char *c;
+                const char *keyspec;
 
                 if (proc_cmdline_value_missing(key, value))
                         return 0;
 
-                r = sscanf(value, "%m[0-9a-fA-F-]=%ms", &uuid, &uuid_value);
-                if (r == 2) {
-                        char *c;
-                        _cleanup_free_ char *keyfile = NULL, *keydev = NULL;
+                n = strspn(value, LETTERS DIGITS "-");
+                if (value[n] != '=') {
+                        if (free_and_strdup(&arg_default_keyfile, value) < 0)
+                                 return log_oom();
+                        return 0;
+                }
 
-                        d = get_crypto_device(uuid);
-                        if (!d)
-                                return log_oom();
+                uuid = strndup(value, n);
+                if (!uuid)
+                        return log_oom();
 
-                        c = strrchr(uuid_value, ':');
-                        if (!c)
-                                /* No keydev specified */
-                                return free_and_replace(d->keyfile, uuid_value);
+                d = get_crypto_device(uuid);
```

Comment:
so, hmm, if i specify LABEL=foobar then this will parse "LABEL" as uuid, no?

Shouldn't there be some check with id128_is_valid() here before we assume it is a crypto device that is specified?

Reply:
Yeah, we don't really do that anywhere else but I figure we should. ```get_crypto_device()``` gives you object that is already in the hashmap or it creates one on the fly and gives you that if it is the first time we see that uuid. I propose we change it ```get_crypto_device()``` so that we don't have to validate uuids in all other places too.

Reply:
@poettering any preference on how to deal with the uuid validity checking?

Reply:
dunno, i'd leave that to you. I'd personally consider this part of "parsing" not "processing", hence i'd do it where we parse, not where we process, i.e. before get_crypt_device(), but quite frankly, it's nothing I feel strongly about...

Another option is do parse the uuid properly, and just pass around sd_id128_t...